### ISSUE TITLE
Improve `Style/ModuleFunction` cop

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12348]
## Description

https://rubystyle.guide/#module-function (already covered by the cop)

I would like to extend the cop (or create a new one) in order to be aware of another rule.

https://rubystyle.guide/#modules-vs-classes

Prefer modules to classes with only class methods. Classes should be used only when it makes sense to create instances out of them.

## Describe the solution you'd like

```ruby
# bad
class SomeClass
  def self.some_method
    # body omitted
  end

  def self.some_other_method
    # body omitted
  end
end

# good
module SomeModule
  module_function

  def some_method
    # body omitted
  end

  def some_other_method
    # body omitted
  end
end
```

**Comment by fatkodima** [comment]
Already implemented via `Style/StaticClass`.

**Comment by ydakuka** [comment]
Agree, my bad.
